src/avram/base_query_template.cr

```diff
@@ -1,11 +1,23 @@
 class Avram::BaseQueryTemplate
-  macro setup(type, columns, associations, table_name, *args, **named_args)
+  macro setup(type, columns, associations, table_name, primary_key_name, *args, **named_args)
     class ::{{ type }}::BaseQuery < Avram::Query
       include Avram::Queryable({{ type }})
 
       @@table_name = :{{ table_name }}
       @@schema_class = {{ type }}
 
+      # If not using default 'id' primary key
+      {% if primary_key_name.id != "id".id %}
+        # Then point 'id' to the primary key
+        def id(*args, **named_args)
```

Comment:
There is an `id` method defined so that things depending on it still work as expected